src/platform/Tizen/PlatformManagerImpl.cpp

```diff
@@ -131,9 +131,17 @@ void PlatformManagerImpl::_Shutdown()
 {
     Internal::GenericPlatformManagerImpl_POSIX<PlatformManagerImpl>::_Shutdown();
 
+    if (mGLibMainLoop == nullptr)
```

Comment:
I would ask why the `Shutdown()` is called twice? If that's a normal behavior, then you will have to move this check at the beginning of this function (above `GenericPlatformManagerImpl_POSIX::_Shutdown()`), because calling  `GenericPlatformManagerImpl_POSIX::_Shutdown()` twice will cause double-free.

However, if we need to protect against calling `Shutdown()` twice in a row, what about `Init()`? It also can not be called twice (without calling Shutdown in between). And this issue is not Tizen specific. IMHO, the rule applies to almost all Init/Shutdown calls.

Reply:
I would ask why the Shutdown() is called twice? 
> It depends on the application's logic. We will request to check it to the application side. But, basically.. that scenario is possible in the current implementation. So, this patchset is to prevent that issue.

For the Init case, I think we need to consider more cases or conditions. So after applying the patchset, will try to consider it.

01-01 09:02:18.083+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DMG] AccessControl: finishing
01-01 09:02:18.083+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DMG] Examples::AccessControlDelegate::Finish
01-01 09:02:18.179+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [FP] Shutting down FabricTable
01-01 09:02:18.179+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [TS] Pending Last Known Good Time: 2023-07-19T10:54:43
01-01 09:02:18.179+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [TS] Previous Last Known Good Time: 2023-07-19T10:54:43
01-01 09:02:18.179+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [TS] Reverted Last Known Good Time to previous value

// Called first time
01-01 09:02:18.179+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] PlatformManagerImpl::_Shutdown called

01-01 09:02:18.179+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] Inet Layer shutdown
01-01 09:02:18.180+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] System Layer shutdown
01-01 09:02:18.180+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] System Layer shutdown done
01-01 09:02:18.180+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] PlatformManagerImpl::_Shutdown done
01-01 09:02:18.180+0900 I/ST-HUB-CORE(P 2930, T 3060): [matter] Resolver.cpp(224) > Resolver::SetState: Received SHUTDOWN event, moving from state ENABLED to DISABLED
01-01 09:02:18.181+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] DNSsd Shutdown: Error: -30015487

// Called second time
01-01 09:02:18.181+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] PlatformManagerImpl::_Shutdown called

01-01 09:02:18.181+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] Inet Layer shutdown
01-01 09:02:18.181+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] System Layer shutdown
01-01 09:02:18.181+0900 E/ST-HUB-CORE(P 2930, T 3060): [matter.sdk] SdkLogging.cpp(63) > [DL] System Layer shutdown done

Reply:
@arkq,

I create the issue to handle the potential issue. After merging this PR, we can continue to follow up remain things.

https://github.com/project-chip/connectedhomeip/issues/28154